examples/crm/src/dataProvider/activity.ts

```diff
@@ -0,0 +1,192 @@
+import { DataProvider, Identifier } from 'react-admin';
+import {
+    COMPANY_CREATED,
+    CONTACT_CREATED,
+    CONTACT_NOTE_CREATED,
+    DEAL_CREATED,
+    DEAL_NOTE_CREATED,
+} from '../consts';
+import {
+    Activity,
+    Company,
+    Contact,
+    ContactNote,
+    Deal,
+    DealNote,
+    Sale,
+} from '../types';
+
+export async function getActivityLog(
+    dataProvider: DataProvider,
+    companyId?: Identifier
+) {
+    const sales = await getSales(dataProvider);
+    const companies = await getCompaniesLog(dataProvider, sales, companyId);
+
+    const [contactsLog, dealsLog] = await Promise.all([
+        getContactsLog(dataProvider, sales, companies),
+        getDealsLog(dataProvider, sales, companies),
+    ]);
+
+    return companies.companiesLog.concat(contactsLog, dealsLog).sort(
+        (a, b) => a.date.localeCompare(b.date) * -1 // sort by date desc
+    );
+}
+
+async function getSales(dataProvider: DataProvider) {
+    const salesDict = await dataProvider
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**nitpick (non-blocking):** More understandable variable name?‏